### PR TITLE
#158765898 [Feature] Setting up GraphQL

### PR DESCRIPTION
**Pull request by makindemayowa**
#### What Does This PR Do?
- lt sets up GraphQL to work side by side with the Django-rest-API

#### Description Of Task To Be Completed
- When that a user hits the `/graphql` endpoint, the user should be able to interact via the browser with the `/graphql` endpoint and via a GraphQL client such as Apollo with the `/graphql` endpoint.

#### Any Background Context You Want To Provide?
- GraphQL helps for faster querying 
- It also reduces the number of API calls the client needs to make
- It helps solve the problem of over-fetching or under-fetching resources.

#### How can this be manually tested?
-  Run `pip install -r server/requirements.txt`
- Navigate to `/graphql` route on your browser to make queries and mutations.

#### What are the relevant github issues?
N/A

#### Screenshot(s)
screen shot 2018-07-10 at 7 55 36 am
screen shot 2018-07-09 at 12 34 37 pm
screen shot 2018-07-10 at 7 59 13 am

### CODE REVIEW

File: server/graphql_schemas/urls.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import rest_framework
+from django.conf.urls import url
+from graphene_django.views import GraphQLView
+from rest_framework.decorators import authentication_classes, permission_classes, api_view
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.settings import api_settings
+
+class DRFAuthenticatedGraphQLView(GraphQLView):
```
Comment (andela-sjames) [inline server/graphql_schemas/urls.py]: This should not be here

File: start [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
Comment (andela-sjames) [inline start]: Why do we have this replicated here?

File: server/graphql_schemas/attend/schema.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import graphene
+from api.models import Attend, Event, GoogleUser
```
Comment (andela-sjames) [inline server/graphql_schemas/attend/schema.py]: Imports should follow PEP should follow PEP8 style guide https://www.python.org/dev/peps/pep-0008/#imports

File: server/graphql_schemas/attend/schema.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        user = info.context.user
+        googleUser = GoogleUser.objects.get(app_user_id=user.id)
+        subscribedEvent = Attend.objects.filter(event_id=event_id, user_id=googleUser.id).first()
+        if not subscribedEvent:
+          raise GraphQLError("The User {0}, has not subscribed to this event".format(user))
+        subscribedEvent.delete()
+        return cls(unsubscribed_event=subscribedEvent)
+
+class Query(object):
+  ####
```
Comment (andela-sjames) [inline server/graphql_schemas/attend/schema.py]: What are we commenting out here

File: server/graphql_schemas/category/schema.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+from api.models import Category
```
Comment (andela-sjames) [inline server/graphql_schemas/category/schema.py]: Imports? 
```
Imports should be grouped in the following order:

Standard library imports.
Related third party imports.
Local application/library specific imports.
```

File: server/graphql_schemas/category/schema.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    model = Category
+    filter_fields = {
+      'name': ['exact', 'icontains', 'istartswith'],
+      'description': ['icontains', 'istartswith'],
+    }
+    interfaces = (relay.Node,)
+
+
+class Query(object):
+  ####
```
Comment (andela-sjames) [inline server/graphql_schemas/category/schema.py]: ??

File: server/graphql_schemas/attend/schema.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
+import graphene
+from api.models import Attend, Event, GoogleUser
+from graphene import relay, ObjectType
+from graphene_django.filter import DjangoFilterConnectionField
+from graphene_django.types import DjangoObjectType
+from graphql import GraphQLError
```
Comment (mbithenzomo) [inline server/graphql_schemas/attend/schema.py]: From https://www.python.org/dev/peps/pep-0008/#imports:

Imports should be grouped in the following order:

Standard library imports.
Related third party imports.
Local application/library specific imports.
You should put a blank line between each group of imports.

File: server/graphql_schemas/event/schema.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+from graphene_django.filter import DjangoFilterConnectionField
+from graphene_django.types import DjangoObjectType
+
+
+class EventNode(DjangoObjectType):
+  class Meta:
+    model = Event
+    filter_fields = {}
+    interfaces = (relay.Node,)
+
```
Comment (andela-sjames) [inline server/graphql_schemas/event/schema.py]: Two lines between classes. ¯\_(ツ)_/¯

File: server/graphql_schemas/interest/schema.py [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    ).first()
+    if not unjoined_social_club:
+      raise GraphQLError("The User {0}, has not joined {1}. ".format(user, category))
+
+    unjoined_social_club.delete()
+    return UnJoinSocialClub(unjoined_social_club=unjoined_social_club)
+
+
+class Query(object):
+  ####
```
Comment (andela-sjames) [inline server/graphql_schemas/interest/schema.py]: ???

**Pull request merged**